File: climbing-stairs/climbing-stairs.py
class Solution:
    def fib(self, n , dp):
        if n <= 1:
            return n
        if dp[n] != -1:
            return dp[n]
        dp[n] = self.fib(n-1,dp) + self.fib(n-2,dp)
        
        return dp[n]
    
    def climbStairs(self,N):
        dp = [0]*(N + 1)
        dp[0] = 1
        dp[1] = 1

        for i in range(2, N + 1):
            dp[i] = dp[i - 1] + dp[i - 2]

        return dp[N]

File: climbing-stairs/climbing-stairs.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let mut one = 1;
        let mut two = 1;
        
        for i in 0..n-1{
            let temp = one;
            one = one + two;
            two = temp;
        }
        
        return one
    }
}